confmap/provider/internal/configurablehttpprovider/provider.go

```diff
@@ -0,0 +1,132 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configurablehttpprovider // import "go.opentelemetry.io/collector/confmap/provider/internal/configurablehttpprovider"
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/internal"
+)
+
+type SchemeType string
+
+const (
+	HTTPScheme  SchemeType = "http"
+	HTTPSScheme SchemeType = "https"
+)
+
+type provider struct {
+	scheme             SchemeType
+	caCertPath         string // Used for tests
+	insecureSkipVerify bool   // Used for tests
+}
+
+// New returns a new provider that reads the configuration from http server using the configured transport mechanism
+// depending on the selected scheme.
+// There are two types of transport supported: PlainText (HTTPScheme) and TLS (HTTPSScheme).
+//
+// One example for http-uri: http://localhost:3333/getConfig
+// One example for https-uri: https://localhost:3333/getConfig
+// This is used by the http and https external implementations.
+func New(scheme SchemeType) confmap.Provider {
+	return &provider{scheme: scheme, caCertPath: "", insecureSkipVerify: false}
```

Comment:
No need to initialize `caCertPath: "", insecureSkipVerify: false` with their zero values.